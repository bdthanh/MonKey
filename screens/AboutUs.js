import React from 'react'
import { Text, View, StyleSheet, TouchableOpacity, Linking } from 'react-native';
import { Octicons, MaterialCommunityIcons } from '@expo/vector-icons';
import { ScreenWidth } from '../components/constants';
import { StatusBarHeight } from '../components/constants';
import { colors } from '../components/colors';
import { ScrollView } from 'react-native';
const { darkBlue, darkYellow } = colors

const AboutUs = ({navigation}) => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <View style={{flex:2, paddingLeft:5, paddingBottom:7}}>
          <TouchableOpacity onPress={()=>navigation.goBack()}>
            <MaterialCommunityIcons name='chevron-left' size={44} color={darkBlue}/>
          </TouchableOpacity>
        </View>
        <View style={{flex:8,alignItems:'center',justifyContent:'center'}}>
          <Text style={styles.boldBlueHeaderText}>About</Text>
        </View>
        <View style={{flex:2}}></View>
      </View>
      <ScrollView style={{paddingHorizontal:15}}>
        <Text style={{fontSize:30, color: '#494949', paddingTop:20, paddingBottom:15, paddingHorizontal:15, fontWeight:'700'}}>Monkey</Text>
        <Text style={{fontSize:17, color: '#494949', fontWeight:'400'}}>
          To know more out MonKey, please refer to:
        </Text>
        <View style={{flexDirection:'row', alignItems:'center'}}>
          <Octicons name='dot-fill' size={14} color='#494949'/>
          <Text style={{fontSize:17, color:darkBlue}} onPress={() => Linking.openURL('http://google.com')}>   Documentation</Text>
        </View>
        <View style={{flexDirection:'row', alignItems:'center'}}>
          <Octicons name='dot-fill' size={14} color='#494949'/>
          <Text style={{fontSize:17, color:darkBlue}} onPress={() => Linking.openURL(' https://github.com/ngquyduc/MonKey')}>   Github repo</Text>
        </View>
      </ScrollView>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex:1,
    backgroundColor: '#fff',
  },
  header: {
    alignItems:'flex-end', 
    justifyContent:'center',
    flexDirection:'row',
    backgroundColor:'#fff',
    borderBottomColor:'#808080',
    borderBottomWidth:1,
    height: StatusBarHeight + 48,
  },
  boldBlueHeaderText: {
    fontSize: 34,
    color: darkBlue,
    marginBottom: 10,
    fontWeight: 'bold',
  },
  ratingView: {
    flexDirection:'column',
    alignItems:'center',
    justifyContent:'center',
    backgroundColor: '#fff',
    borderRadius:25,
    marginBottom:15,
    shadowColor:'#999',
    shadowOffset: {width:0,height:1},
    shadowOpacity:0.8,
    shadowRadius:2,
    elevation:5,
    marginHorizontal:20,
    paddingVertical:30,
    marginTop:20,
  },
  feedbackView: {
    flexDirection:'column',
    alignItems:'center',
    backgroundColor: '#fff',
    borderRadius:25,
    marginBottom:10,
    shadowColor:'#999',
    shadowOffset: {width:0,height:1},
    shadowOpacity:0.8,
    shadowRadius:2,
    elevation:5,
    marginHorizontal:20,
    height:300,
    padding:10
  },
  noteView: {
    borderRadius:20,
    backgroundColor:'#f1fbfd',
    shadowColor:'#999',
    shadowOffset: {width:0,height:1},
    shadowOpacity:0.8,
    shadowRadius:2,
    elevation:-3,
    height:243,
    width:ScreenWidth-70
  },
  submitButtonView: {
    marginTop: 10, 
    alignItems:'center',
    justifyContent:'center',
    paddingBottom:4,
    paddingTop:4,
    paddingLeft:4,
    borderBottomColor: '#E9E9E9',
    borderTopColor: '#E9E9E9',     
    height:48
  },
  cancelText: {
    fontSize:18,
    color:'#fff',
    fontWeight:'500',
  },
  inputButton: {
    padding: 5,
    flexDirection: 'row',
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius:10,
    backgroundColor:darkYellow,
    width:120
  },
})
export default AboutUs;