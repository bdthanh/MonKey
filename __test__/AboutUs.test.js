import React from 'react';
import {create} from 'react-test-renderer'
import AboutUs from '../screens/AboutUs';

const tree = create(<AboutUs/>);

test('snapshot AboutUs', () => {
  expect(tree).toMatchSnapshot();
});
