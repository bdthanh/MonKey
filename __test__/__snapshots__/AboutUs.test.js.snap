// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot AboutUs 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#fff",
        "borderBottomColor": "#808080",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 62,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 2,
          "paddingBottom": 7,
          "paddingLeft": 5,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#35baf6",
            "fontSize": 34,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        About
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 2,
        }
      }
    />
  </View>
  <RCTScrollView
    style={
      Object {
        "paddingHorizontal": 15,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#494949",
            "fontSize": 30,
            "fontWeight": "700",
            "paddingBottom": 15,
            "paddingHorizontal": 15,
            "paddingTop": 20,
          }
        }
      >
        Monkey
      </Text>
      <Text
        style={
          Object {
            "color": "#494949",
            "fontSize": 17,
            "fontWeight": "400",
            "paddingBottom": 10,
            "paddingHorizontal": 5,
          }
        }
      >
        MonKey is a mobile application providing a big-picture view of your finances, which easily help you to keep track of your money flow.
      </Text>
      <Text
        style={
          Object {
            "color": "#494949",
            "fontSize": 17,
            "fontWeight": "400",
            "paddingHorizontal": 5,
          }
        }
      >
        To know more about MonKey, please refer to:
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingHorizontal": 5,
          }
        }
      >
        <Text />
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#35baf6",
              "fontSize": 17,
            }
          }
        >
             Documentation
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingHorizontal": 5,
          }
        }
      >
        <Text />
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#35baf6",
              "fontSize": 17,
            }
          }
        >
             Github repo
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#494949",
            "fontSize": 30,
            "fontWeight": "700",
            "paddingBottom": 15,
            "paddingHorizontal": 15,
            "paddingTop": 20,
          }
        }
      >
        Developers
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 25,
            "elevation": 5,
            "flexDirection": "column",
            "marginBottom": 15,
            "padding": 15,
            "shadowColor": "#999",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.7,
            "shadowRadius": 4,
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomColor": "#e9e9e9",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "paddingBottom": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#494949",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "paddingBottom": 6,
                }
              }
            >
              Bui Duc Thanh
            </Text>
            <Text
              style={
                Object {
                  "color": "#979797",
                  "fontSize": 15,
                  "fontWeight": "300",
                }
              }
            >
              National University of Singapore
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#6200ee",
                    "borderRadius": 25,
                    "height": 50,
                    "width": 50,
                  },
                  undefined,
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "borderRadius": 25,
                    "height": 50,
                    "width": 50,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 10,
            }
          }
        >
          <Text />
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#494949",
                "fontSize": 17,
              }
            }
          >
            bdthanh@u.nus.edu
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 10,
            }
          }
        >
          <Text />
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#494949",
                "fontSize": 17,
              }
            }
          >
            bdthanh
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text />
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#494949",
                "fontSize": 17,
              }
            }
          >
            buiducthanh2003
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 25,
            "elevation": 5,
            "flexDirection": "column",
            "marginBottom": 15,
            "padding": 15,
            "shadowColor": "#999",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.7,
            "shadowRadius": 4,
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomColor": "#e9e9e9",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "paddingBottom": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#494949",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "paddingBottom": 6,
                }
              }
            >
              Nguyen Quy Duc
            </Text>
            <Text
              style={
                Object {
                  "color": "#979797",
                  "fontSize": 15,
                  "fontWeight": "300",
                }
              }
            >
              National University of Singapore
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#6200ee",
                    "borderRadius": 25,
                    "height": 50,
                    "width": 50,
                  },
                  undefined,
                ]
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "borderRadius": 25,
                    "height": 50,
                    "width": 50,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 10,
            }
          }
        >
          <Text />
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#494949",
                "fontSize": 17,
              }
            }
          >
            quyduc@u.nus.edu
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 10,
            }
          }
        >
          <Text />
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#494949",
                "fontSize": 17,
              }
            }
          >
            ngquyduc
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text />
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#494949",
                "fontSize": 17,
              }
            }
          >
            quyduc
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
